pkg/logql/shardmapper_test.go

```diff
@@ -1367,8 +1454,8 @@ func TestMapping(t *testing.T) {
 			mapped, _, err := m.Map(ast, nilShardMetrics.downstreamRecorder())
 
 			require.Equal(t, tc.err, err)
-			require.Equal(t, tc.expr.String(), mapped.String())
-			require.Equal(t, tc.expr, mapped)
+			require.Equal(t, mapped.String(), tc.expr.String())
+			require.Equal(t, mapped, tc.expr)
```

Comment:
Changed order of the arguments to get correct hints in test output if it fails. First argument is the expected value, second argument is the actual value.

Reply:
yes but `tc.expr` is the expected.